core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -319,14 +331,54 @@ object Extraction {
       }
     }
 
+    def newTuple(root: JValue, mappings: List[Mapping]): Any = {
+      root match {
+        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+          val builtItems: Seq[Object] = items.zip(mappings).map({
+            case (i,m) =>
+              build(i, m).asInstanceOf[Object]
+          })
+          val tupleClass = Class.forName("scala.Tuple" + items.length)
```

Comment:
We should build this ahead of time in a top-level `final val`; `Class.forName` is really expensive. We might also use that to remove the `>=1 && <= 22` magic numbers above.

This could go hand-in-hand with our friendly neighborhood `tuples` set, since it's already got the right `Class` instance anyway.